### DISCUSSION TITLE
Could not establish connection. Receiving end does not exist when extension updated on store

### DISCUSSION THREAD

**Discussion by Anyasya** [wxt-dev/wxt#856] [Q&A]
### Describe the bug

i know this error has always been around and it's often caused by an architectural deficiency in Chrome - the lack of automatic re-injection of the content scripts in the currently open tabs after the extension is updated/installed/reloaded like Firefox does it. The solution for Chrome is to inject the scripts explicitly in chrome.runtime.onInstalled but i don't know how to do it in wxt 

my background.ts

```
import { onMessage, sendMessage } from '@/helpers/messaging';

export default defineBackground(() => {
  onMessage('popup:getData', async (message) => {
    return await sendMessage(
      'background:getData',
      undefined,
      message.data,
    );
  });
});
```

### Reproduction

Could not establish connection. Receiving end does not exist when extension updated on store

### System Info

```shell
Chrome
```

### Used Package Manager

npm

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/wxt-dev/wxt/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://wxt.dev/guide/installation.html).
- [X] Check that there isn't [already an issue](https://github.com/wxt-dev/wxt/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/wxt-dev/wxt/discussions) or join our [Discord Chat Server](https://discord.gg/ZFsZqGery9).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by aklinker1** [comment]
this is more of a question than a bug? I'm gonna convert this to a discussion.

**Comment by aklinker1** [comment]
WXT doesn't do this automatically for you, since it requires additional permissions. See this stack overflow answer instead: 

https://stackoverflow.com/questions/10994324/chrome-extension-content-script-re-injection-after-upgrade-or-install#11598753
